src/components/Badge/index.tsx

```diff
@@ -1,167 +1,152 @@
+    `background${definedProminence}` as keyof typeof theme.colors.primary
+
+  return {
+    ...SENTIMENTS.reduce(
+      (reducer, sentiment) => ({
+        ...reducer,
+        [sentiment]: `
+      color: ${theme.colors[sentiment][text]};
+      background: ${theme.colors[sentiment][background]}
+    `,
+      }),
+      {},
+    ),
+    disabled: `
+      color: ${theme.colors.neutral.textWeak};
+      background: ${theme.colors.neutral.backgroundStrong}
+    `,
+    neutral: `
+      color: ${
+        prominence === PROMINENCES.strong
+          ? theme.colors.neutral[text]
+          : theme.colors.neutral.textWeak
+      };
+      background: ${theme.colors.neutral[background]}
+    `,
+  }
+}
 
 type BadgeProps = {
-  variant?: Variants
-  size?: Sizes
-  children: ReactNode
-} & BoxProps
+  variant?: Color
+  size?: keyof typeof SIZES
+  prominence?: keyof typeof PROMINENCES
+  icon?: IconName
+  disabled?: boolean
+  className?: string
+}
 
-const StyledBox = styled(Box, {
+const StyledBox = styled('div', {
   shouldForwardProp: prop => !['variant', 'size'].includes(prop.toString()),
-})<BadgeProps>`
+})<{ size: number; variant: string }>`
   display: inline-flex;
   align-items: center;
   justify-content: center;
   white-space: nowrap;
   border-radius: ${({ theme }) => theme.space['2']};
-  padding: 0 ${({ theme }) => theme.space['2']};
+  padding: 0
+    ${({ theme, size }) =>
+      size === SIZES.small ? theme.space['1'] : theme.space['2']};
+  gap: ${({ theme, size }) =>
+    size === SIZES.small ? theme.space['0.5'] : theme.space['1']};
   width: fit-content;
-
-  ${variantsStyle}
-  ${sizesStyle}
+  height: ${({ size }) => size}px;
+  ${({ variant }) => variant}
 `
 
-const Badge = ({
+const Badge: FunctionComponent<BadgeProps> = ({
   variant = 'neutral',
   size = 'medium',
-  ...props
-}: BadgeProps) => {
+  prominence = 'default',
+  icon,
+  disabled = false,
+  className,
+  children,
+}) => {
+  const theme = useTheme()
+
   /**
    * Badge should display an aria-label if the status is not neutral or primary
    */
-  const ariaLabel = useMemo(() => {
-    const strippedVariant = variant.replace('light-', '')
+  const ariaLabel = useMemo(
+    () =>
+      ['neutral', 'primary'].some(baseVariant => baseVariant === variant)
+        ? undefined
+        : variant,
+    [variant],
+  )
 
-    return ['neutral', 'primary'].some(
-      baseVariant => baseVariant === strippedVariant,
-    )
-      ? undefined
-      : strippedVariant
-  }, [variant])
+  const generatedStyles = useMemo(
+    () => generateStyles({ prominence, theme }),
+    [prominence, theme],
+  )
 
   return (
     <StyledBox
       role="status"
       aria-label={ariaLabel}
       as="span"
-      variant={variant}
-      size={size}
-      {...props}
-    />
+      variant={disabled ? generatedStyles.disabled : generatedStyles[variant]}
+      size={SIZES[size]}
+      className={className}
+    >
+      {icon ? <Icon name={icon} size={12} /> : null}
```

Comment:
fixed size? Icon doesn't change if text/badge is a bit bigger ?

Reply:
and i find 12 extremely tiny

Reply:
<img width="145" alt="Screenshot 2022-04-20 at 08 50 18" src="https://user-images.githubusercontent.com/3360430/164167851-0041e745-5cdc-44b3-bf2b-6b88c33acec0.png">


We should at least vertical align icon

Reply:
@vincentaudebert icon is fixed size on design specs and is already vertically aligned with text you can see it when inspecting code. I'm gonna check with design team if size is correct

Reply:
@vincentaudebert I've seen it with design team, you are right about the icon. I removed it for small badge and made it 16px when size is medium or large. It should be better now  :)